Description: upstream: source: C2help2man
 Attempt to format and harmonize the help and version messages
 of the iprange front-end program wrt UN*X customs in view to
 employ help2man to generate a manpage; meant to be submitted
 to the upstream maintainer.
Origin: debian
Author: Jerome Benoit <calculus@rezozer.net>
Last-Update: 2015-12-27

--- a/iprange.c
+++ b/iprange.c
@@ -2102,288 +2102,268 @@
  */
 
 static void usage(const char *me) {
-	fprintf(stderr, "\n"
-		"iprange\n"
-		"manage IP ranges\n"
-		"version: " VERSION "\n"
-		"\n"
-		"Original,   Copyright (C) 2003 Gabriel L. Somlo\n"
-		"Adapted,    Copyright (C) 2004 Paul Townsend\n"
-		"Refactored, Copyright (C) 2015 Costa Tsaousis for FireHOL\n"
-		"License: GPL\n"
+	fprintf(stdout,
+		"iprange manages IP ranges\n"
 		"\n"
 		"Usage: %s [options] file1 file2 file3 ...\n"
 		"\n"
-		"options (multiple options are aliases):\n"
-		"\n"
-		"	--------------------------------------------------------------\n"
-		"	CIDR OUTPUT MODES\n"
+		"Options:\n"
+		"multiple options are aliases\n"
 		"\n"
+		"CIDR output modes:\n"
 		"	--optimize\n"
 		"	--combine\n"
 		"	--merge\n"
 		"	--union\n"
 		"	-J\n"
 		"		> MERGE mode (the default)\n"
-		"		returns all IPs found on all files\n"
-		"		the resulting set is sorted\n"
+		"		Returns all IPs found on all files.\n"
+		"		The resulting set is sorted.\n"
 		"\n"
 		"	--common\n"
 		"	--intersect\n"
 		"		> COMMON mode\n"
-		"		intersect all files to find their common IPs\n"
-		"		the resulting set is sorted\n"
+		"		Intersect all files to find their common IPs.\n"
+		"		The resulting set is sorted.\n"
 		"\n"
 		"	--except\n"
 		"	--exclude-next\n"
 		"		> EXCEPT mode\n"
-		"		here is how it works:\n"
-		"		1. merge all files before this parameter (ipset A)\n"
-		"		2. remove all IPs found in the files after this\n"
-		"		   parameter, from ipset A and print what remains\n"
-		"		the resulting set is sorted\n"
+		"		Here is how it works:\n"
+		"		(1) merge all files before this parameter (ipset A);\n"
+		"		(2) remove all IPs found in the files after this\n"
+		"		parameter, from ipset A and print what remains.\n"
+		"		The resulting set is sorted.\n"
 		"\n"
 		"	--diff\n"
 		"	--diff-next\n"
 		"		> DIFF mode\n"
-		"		here is how it works:\n"
-		"		1. merge all files before this parameter (ipset A)\n"
-		"		2. merge all files after this parameter (ipset B)\n"
-		"		3. print all differences between A and B, i.e IPs\n"
-		"		   found is either A or B, but not both\n"
-		"		the resulting set is sorted\n"
-		"		when there are differences, iprange exits with 1\n"
-		"		when ipsets A and B are the same, exits with 0"
+		"		Here is how it works:\n"
+		"		(1) merge all files before this parameter (ipset A);\n"
+		"		(2) merge all files after this parameter (ipset B);\n"
+		"		(3) print all differences between A and B, i.e IPs\n"
+		"		found is either A or B, but not both.\n"
+		"		The resulting set is sorted.\n"
+		"		When there are differences between A and B, iprange\n"
+		"		exits with 1, with 0 otherwise."
 		"\n"
 		"	--ipset-reduce PERCENT\n"
 		"	--reduce-factor PERCENT\n"
 		"		> IPSET REDUCE mode\n"
-		"		merge all files and print the merged set\n"
+		"		Merge all files and print the merged set,\n"
 		"		but try to reduce the number of prefixes (subnets)\n"
-		"		found, while allowing some increase in entries\n"
-		"		the PERCENT is how much percent to allow\n"
-		"		increase on the number of entries in order to reduce\n"
+		"		found, while allowing some increase in entries.\n"
+		"		The PERCENT is how much percent to allow increase\n"
+		"		on the number of entries in order to reduce\n"
 		"		the prefixes (subnets)\n"
-		"		(the internal default PERCENT is 20)\n"
-		"		(use -v to see exactly what it does)\n"
-		"		the resulting set is sorted\n"
+		"		(the internal default PERCENT is 20).\n"
+		"		Use -v to see exactly what it does.\n"
+		"		The resulting set is sorted.\n"
 		"\n"
 		"	--ipset-reduce-entries ENTRIES\n"
 		"	--reduce-entries ENTRIES\n"
 		"		> IPSET REDUCE mode\n"
-		"		allow increasing the entries above PERCENT, if\n"
-		"		they are below ENTRIES\n"
-		"		(the internal default ENTRIES is 16384)\n"
-/*
+		"		Allow increasing the entries above PERCENT,\n"
+		"		if they are below ENTRIES\n"
+		"		(the internal default ENTRIES is 16384).\n"
+#if 0
 		"\n"
 		"	--histogram\n"
 		"		> IPSET HISTOGRAM mode\n"
-		"		maintain histogram data for ipset and dump current\n"
-		"		status\n"
+		"		Maintain histogram data for ipset and\n"
+		"   dump current status.\n"
 		"\n"
 		"	--histogram-dir PATH\n"
 		"		> IPSET HISTOGRAM mode\n"
-		"		the directory to keep histogram data\n"
-*/
-		"\n"
+		"		Specify where to keep histogram data.\n"
+#endif
 		"\n"
-		"	--------------------------------------------------------------\n"
-		"	CSV OUTPUT MODES\n"
 		"\n"
+		"CSV output modes:\n"
 		"	--compare\n"
-		"		> COMPARE ALL mode (CSV output)\n"
-		"		compare all files with all other files\n"
-		"		add --header to get the CSV header too\n"
+		"		> COMPARE ALL mode\n"
+		"		Compare all files with all other files.\n"
+		"		Add --header to get the CSV header too.\n"
 		"\n"
 		"	--compare-first\n"
-		"		> COMPARE FIRST mode (CSV output)\n"
-		"		compare the first file with all other files\n"
-		"		add --header to get the CSV header too\n"
+		"		> COMPARE FIRST mode\n"
+		"		Compare the first file with all other files.\n"
+		"		Add --header to get the CSV header too.\n"
 		"\n"
 		"	--compare-next\n"
-		"		> COMPARE NEXT mode (CSV output)\n"
-		"		compare all the files that appear before this\n"
+		"		> COMPARE NEXT mode\n"
+		"		Compare all the files that appear before this\n"
 		"		parameter, to all files that appear after this\n"
-		"		parameter\n"
-		"		add --header to get the CSV header too\n"
+		"		parameter.\n"
+		"		Add --header to get the CSV header too.\n"
 		"\n"
 		"	--count-unique\n"
 		"	-C\n"
-		"		> COUNT UNIQUE mode (CSV output)\n"
-		"		merge all files and print its counts\n"
-		"		add --header to get the CSV header too\n"
+		"		> COUNT UNIQUE mode\n"
+		"		Merge all files and print its counts.\n"
+		"		Add --header to get the CSV header too.\n"
 		"\n"
 		"	--count-unique-all\n"
-		"		> COUNT UNIQUE ALL mode (CSV output)\n"
-		"		print counts for each file\n"
-		"		add --header to get the CSV header too\n"
-		"\n"
+		"		> COUNT UNIQUE ALL mode\n"
+		"		Print counts for each file.\n"
+		"		Add --header to get the CSV header too.\n"
 		"\n"
-		"	--------------------------------------------------------------\n"
-		"	OPTIONS THAT AFFECT INPUT\n"
 		"\n"
+		"Controlling input:\n"
 		"	--dont-fix-network\n"
-		"		by default, the network address of all CIDRs\n"
-		"		is used (i.e. 1.1.1.17/24 is read as 1.1.1.0/24)\n"
+		"		By default, the network address of all CIDRs\n"
+		"		is used (i.e., 1.1.1.17/24 is read as 1.1.1.0/24):\n"
 		"		this option disables this feature\n"
-		"		(i.e. 1.1.1.17/24 is read as 1.1.1.17-1.1.1.255)\n"
+		"		(i.e., 1.1.1.17/24 is read as 1.1.1.17-1.1.1.255).\n"
 		"\n"
 		"	--default-prefix PREFIX\n"
 		"	-p PREFIX\n"
 		"		Set the default prefix for all IPs without mask\n"
-		"		the default is 32\n"
+		"		(the default is 32).\n"
 		"\n"
 		"\n"
-		"	--------------------------------------------------------------\n"
-		"	OPTIONS THAT AFFECT CIDR OUTPUT\n"
-		"\n"
+		"Controlling CIDR output:\n"
 		"	--min-prefix N\n"
-		"		do not generate prefixes larger than N\n"
-		"		i.e. if N is 24 then /24 to /32 entries will be\n"
-		"		     generated (a /16 network will be generated\n"
-		"		     using multiple /24 networks)\n"
-		"		this is useful to optimize netfilter/iptables\n"
-		"		ipsets, where each different prefix increases the\n"
-		"		lookup time for each packet, but the number of\n"
-		"		entries in the ipset do not affect its performance\n"
-		"		with this setting more entries will be produced\n"
-		"		to accomplish the same match\n"
-		"		warning: misuse of this parameter can create a large\n"
-		"		         number of entries in the generated set\n"
+		"		Do not generate prefixes larger than N,\n"
+		"		i.e., if N is 24 then /24 to /32 entries will be\n"
+		"		generated (a /16 network will be generated\n"
+		"		using multiple /24 networks).\n"
+		"		This is useful to optimize netfilter/iptables\n"
+		"		ipsets where each different prefix increases the\n"
+		"		lookup time for each packet whereas the number of\n"
+		"		entries in the ipset do not affect its performance.\n"
+		"		With this setting more entries will be produced\n"
+		"		to accomplish the same match.\n"
+		"		WARNING: misuse of this parameter can create a large\n"
+		"		number of entries in the generated set.\n"
 		"\n"
 		"	--prefixes N,N,N, ...\n"
-		"		enable only the given prefixes to express all CIDRs\n"
-		"		prefix 32 is always enabled\n"
-		"		warning: misuse of this parameter can create a large\n"
-		"		         number of entries in the generated set\n"
+		"		Enable only the given prefixes to express all CIDRs;\n"
+		"		prefix 32 is always enabled.\n"
+		"		WARNING: misuse of this parameter can create a large\n"
+		"		number of entries in the generated set.\n"
 		"\n"
 		"	--print-ranges\n"
 		"	-j\n"
-		"		print IP ranges (A.A.A.A-B.B.B.B)\n"
-		"		the default is to print CIDRs (A.A.A.A/B)\n"
-		"		it only applies when the output is not CSV\n"
+		"		Print IP ranges (A.A.A.A-B.B.B.B)\n"
+		"		(the default is to print CIDRs (A.A.A.A/B)).\n"
+		"		It only applies when the output is not CSV.\n"
 		"\n"
 		"	--print-single-ips\n"
 		"	-1\n"
-		"		print single IPs\n"
+		"		Print single IPs;\n"
 		"		this can produce large output\n"
-		"		the default is to print CIDRs (A.A.A.A/B)\n"
-		"		it only applies when the output is not CSV\n"
+		"		(the default is to print CIDRs (A.A.A.A/B)).\n"
+		"		It only applies when the output is not CSV.\n"
 		"\n"
 		"	--print-binary\n"
-		"		print binary data\n"
-		"		this is the fastest way to print a large ipset\n"
-		"		the result can be read by iprange on the same\n"
-		"		architecture (no conversion of endianness)\n"
+		"		Print binary data:\n"
+		"		this is the fastest way to print a large ipset.\n"
+		"		The result can be read by iprange on the same\n"
+		"		architecture (no conversion of endianness).\n"
 		"\n"
 		"	--print-prefix STRING\n"
-		"		print STRING before each IP, range or CIDR\n"
-		"		this sets both --print-prefix-ips and\n"
-		"		--print-prefix-nets\n"
+		"		Print STRING before each IP, range or CIDR.\n"
+		"		This sets both --print-prefix-ips and\n"
+		"		--print-prefix-nets .\n"
 		"\n"
 		"	--print-prefix-ips STRING\n"
-		"		print STRING before each single IP\n"
+		"		Print STRING before each single IP:\n"
 		"		useful for entering single IPs to a different\n"
-		"		ipset than the networks\n"
+		"		ipset than the networks.\n"
 		"\n"
 		"	--print-prefix-nets STRING\n"
-		"		print STRING before each range or CIDR\n"
+		"		Print STRING before each range or CIDR:\n"
 		"		useful for entering sunbets to a different\n"
-		"		ipset than single IPs\n"
+		"		ipset than single IPs.\n"
 		"\n"
 		"	--print-suffix STRING\n"
-		"		print STRING after each IP, range or CIDR\n"
-		"		this sets both --print-suffix-ips and\n"
-		"		--print-suffix-nets\n"
+		"		Print STRING after each IP, range or CIDR.\n"
+		"		This sets both --print-suffix-ips and\n"
+		"		--print-suffix-nets .\n"
 		"\n"
 		"	--print-suffix-ips STRING\n"
-		"		print STRING after each single IP\n"
+		"		Print STRING after each single IP:\n"
 		"		useful for giving single IPs different\n"
-		"		ipset options\n"
+		"		ipset options.\n"
 		"\n"
 		"	--print-suffix-nets STRING\n"
-		"		print STRING after each range or CIDR\n"
+		"		Print STRING after each range or CIDR:\n"
 		"		useful for giving subnets different\n"
-		"		ipset options\n"
+		"		ipset options.\n"
 		"\n"
 		"	--quiet\n"
-		"		do not print the actual ipset\n"
-		"		can only be used in DIFF mode\n"
-		"\n"
+		"		Do not print the actual ipset.\n"
+		"		Can only be used in DIFF mode.\n"
 		"\n"
-		"	--------------------------------------------------------------\n"
-		"	OPTIONS THAT AFFECT CSV OUTPUT\n"
 		"\n"
+		"Controlling CSV output:\n"
 		"	--header\n"
-		"		when the output is CSV, print the header line\n"
-		"		the default is to not print the header line\n"
+		"		When the output is CSV, print the header line\n"
+		"		(the default is to not print the header line).\n"
 		"\n"
 		"\n"
-		"	--------------------------------------------------------------\n"
-		"	OPTIONS THAT AFFECT DNS RESOLUTION\n"
-		"\n"
+		"Controlling DNS resolution:\n"
 		"	--dns-threads NUMBER\n"
-		"		the number of parallel DNS queries to execute\n"
+		"		The number of parallel DNS queries to execute\n"
 		"		when the input files contain hostnames\n"
-		"		the default is %d\n"
+		"		(the default is %d).\n"
 		"\n"
 		"	--dns-silent\n"
-		"		do not print DNS resolution errors\n"
-		"		the default is to print all DNS related errors\n"
+		"		Do not print DNS resolution errors\n"
+		"		(the default is to print all DNS related errors).\n"
 		"\n"
 		"	--dns-progress\n"
-		"		print DNS resolution progress bar\n"
-		"\n"
+		"		Print DNS resolution progress bar.\n"
 		"\n"
-		"	--------------------------------------------------------------\n"
-		"	OTHER OPTIONS\n"
 		"\n"
+		"Other options:\n"
 		"	--has-compare\n"
 		"	--has-reduce\n"
-		"		exits with 0\n"
-		"		other versions of iprange will exit with 1\n"
-		"		use this option in scripts to find if this\n"
-		"		version of iprange is present in a system\n"
+		"		Exits with 0,\n"
+		"		other versions of iprange will exit with 1.\n"
+		"		Use this option in scripts to find if this\n"
+		"		version of iprange is present in a system.\n"
 		"\n"
 		"	-v\n"
-		"		be verbose on stderr\n"
+		"		Be verbose on stderr.\n"
 		"\n"
+		"\n"
+		"Getting help:\n"
 		"	--version\n"
-		"		print version\n"
+		"		Print version and exit.\n"
 		"\n"
 		"	--help\n"
 		"	-h\n"
-		"		print this message\n"
-		"\n"
+		"		Print this message and exit.\n"
 		"\n"
-		"	--------------------------------------------------------------\n"
-		"	INPUT FILES\n"
 		"\n"
-		"	fileN\n"
-		"		a filename or - for stdin\n"
-		"		each filename can be followed by [as NAME]\n"
-		"		to change its name in the CSV output\n"
-		"\n"
-		"		if no filename is given, stdin is assumed\n"
-		"\n"
-		"		files may contain any or all of the following:\n"
-		"		- comments starting with # or ;\n"
-		"		- one IP per line (without mask)\n"
-		"		- a CIDR per line (A.A.A.A/B)\n"
-		"		- an IP range per line (A.A.A.A - B.B.B.B)\n"
-		"		- a CIDR range per line (A.A.A.A/B - C.C.C.C/D)\n"
-		"		  the range is calculated as the network address of\n"
-		"		  A.A.A.A/B to the broadcast address of C.C.C.C/D\n"
-		"		  (this is affected by --dont-fix-network)\n"
-		"		- CIDRs can be given in either prefix or netmask\n"
-		"		  format in all cases (including ranges)\n"
-		"		- one hostname per line, to be resolved with DNS\n"
-		"		  (if the IP resolves to multiple IPs, all of them\n"
-		"		  will be added to the ipset)\n"
-		"		  hostnames cannot be given as ranges\n"
-		"		- spaces and empty lines are ignored\n"
+		"Files:\n"
+		"Input files:\n"
+		"	> fileN\n"
+		"		A filename or - for stdin.\n"
+		"		Each filename can be followed by [as NAME]\n"
+		"		to change its name in the CSV output.\n"
+		"		If no filename is given, stdin is assumed.\n"
+		"\n"
+		"		Files may contain any or all of the following:\n"
+		"		(1) comments starting with hashes (#) or semicolons (;);\n"
+		"		(2) one IP per line (without mask);\n"
+		"		(3) a CIDR per line (A.A.A.A/B);\n"
+		"		(4) an IP range per line (A.A.A.A - B.B.B.B);\n"
+		"		(5) a CIDR range per line (A.A.A.A/B - C.C.C.C/D);\n"
+		"		the range is calculated as the network address of\n"
+		"		A.A.A.A/B to the broadcast address of C.C.C.C/D\n"
+		"		(this is affected by --dont-fix-network);\n"
+		"		(6) CIDRs can be given in either prefix or netmask\n"
+		"		format in all cases (including ranges);\n"
+		"		(7) one hostname per line, to be resolved with DNS\n"
+		"		(if the IP resolves to multiple IPs, all of them\n"
+		"		will be added to the ipset)\n"
+		"		hostnames cannot be given as ranges;\n"
+		"		(8) spaces and empty lines are ignored.\n"
 		"\n"
-		"		any number of files can be given\n"
+		"		Any number of files can be given.\n"
 		"\n"
 		, me, dns_threads_max);
 	exit(1);
@@ -2398,7 +2378,17 @@
  */
 
 static void version() {
-	fprintf(stdout, "%s\n", VERSION);
+	fprintf(stdout,
+		"iprange " VERSION "\n"
+		"Copyright (C) 2015 Costa Tsaousis for FireHOL (Refactored)\n"
+		"Copyright (C) 2004 Paul Townsend (Adapted)\n"
+		"Copyright (C) 2003 Gabriel L. Somlo (Original)\n"
+		"\n"
+		"License: GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>.\n"
+		"This program comes with ABSOLUTELY NO WARRANTY; This is free software, and\n"
+		"you are welcome to redistribute it under certain conditions;\n"
+		"See COPYING distributed in the source for details.\n"
+		);
 	exit(1);
 }
 
